snapcraft/application.py

```diff
@@ -185,15 +207,13 @@ def _get_dispatcher(self) -> craft_cli.Dispatcher:
         # project's base (if any). Here, we handle the case where there *is*
         # a project and it's core24, which means it should definitely fall into
         # the craft-application-based flow.
-        if self._snapcraft_yaml_path:
-            with self._snapcraft_yaml_path.open() as file:
-                yaml_data = util.safe_yaml_load(file)
-            base = yaml_data.get("base")
-            build_base = yaml_data.get("build-base")
+        if self._snapcraft_yaml_data:
+            base = self._snapcraft_yaml_data.get("base")
+            build_base = self._snapcraft_yaml_data.get("build-base")
             _get_esm_error_for_base(base)
-            if "core24" in (base, build_base) or build_base == "devel":
-                # We know for sure that we're handling a core24 project
-                self._known_core24 = True
+
+            if self._known_craft_app_base:
```

Comment:
can't you remove this?

Reply:
Not without refactoring this if/else structure, otherwise `core24` snaps would hit the last else statement.

```python
else:
    raise errors.ClassicFallback()
```

I'm not a huge fan of this `pass` or `raise ClassicFallback` design, but I can tweak it to make a little more readable.